src/3rd_party_bundle_remove.c

```diff
@@ -93,9 +93,37 @@ static bool parse_options(int argc, char **argv)
 	return true;
 }
 
-static enum swupd_code remove_bundle_binaries(struct list *removed_files)
+static enum swupd_code remove_bundle_binaries(struct list *removed_files, struct list *common_files)
 {
-	return third_party_process_files(removed_files, "\nRemoving 3rd-party bundle binaries...\n", "remove_binaries", third_party_remove_binary);
+	enum swupd_code ret = SWUPD_OK;
+	char *script = NULL;
+	struct list *iter = NULL;
+	struct file *file = NULL;
+
+	/* remove scripts associated with all the binaries removed */
+	ret = third_party_process_files(removed_files, "\nRemoving 3rd-party bundle binaries...\n", "remove_binaries", third_party_remove_binary);
+
+	/* some binary files from the removed bundle may not have been removed due to
+	 * being needed by bundles still installed in the system, however, if
+	 * the bundles that are still installed didn't export those binaries then
+	 * the scripts from those binaries must be deleted */
+	for (iter = common_files; iter; iter = iter->next) {
+		file = iter->data;
+		if (!file->is_file || !is_binary(file->filename) || file->is_exported) {
+			continue;
+		}
+		/* if the file is exported, we can unexport it now */
```

Comment:
I think the comment should read "if the file isn't exported".

Reply:
Hmm, I see now how that message could be confusing, the file is really exported because at that point of time a script exists that exports that file (and that is what I meant), but it should no longer be exported because the remaining bundles don't have the is_exported flag set (which I believe is what you mean't).

I will rephrase the message so it is not confusing.

Reply:
Ah okay, your explanation makes perfect sense with that context, thanks!